### PR TITLE
Use RVM for rubygem-katello master release

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemRPMNightlyRelease.groovy [hunk 1]
```diff
@@ -32,7 +32,7 @@ pipeline {
             steps {
                 dir('gem-build') {
                     git(url: "https://github.com/${owner_repo}", branch: branch)
-                    sh("gem build ${gemspec}")
+                    withRVM(["gem build ${gemspec}"], '2.5')
```
[the remark this answers is not in the document]
Comment (ehelms) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemRPMNightlyRelease.groovy]: What is magic about this?

[the remark this answers is not in the document]
Comment (ekohl) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemRPMNightlyRelease.groovy]: I'd prefer to switch the default from 2.0 to 2.5 and just use the default here.

[the remark this answers is not in the document]
Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemRPMNightlyRelease.groovy]: what is `2.5`?

[the remark this answers is not in the document]
Comment (ehelms) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemRPMNightlyRelease.groovy]: The default switch is a much bigger change. I'd prefer not to do that here and file an issue somewhere.

@komidore64 You are essentially arguing why do method parameters exist the way they do which is a general topic IMO. I am passing an argument to a function that it exists. To change this I have to change the method signature to a different style that accepts a dictionary which I think is outside the scope of this change.

[the remark this answers is not in the document]
Comment (ekohl) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemRPMNightlyRelease.groovy]: @ehelms I do remember from my university days that my professor would heavily deduct points for magic strings inside programs. He stated that you should introduce a constant like `RUBY_VERSION = '2.5'` and then use `withRVM(..., RUBY_VERSION)` and `cleanupRVM(RUBY_VERSION)`. The benefit is that you give meaning to the `'2.5'`. Another one is that you can't forget changing it in some place. It also improves refactoring in cases where you would have multiple `2.5` strings (say Katello would be version `2.5` as well). Then there's a risk of changing too much or too little. I assume @komidore64 is referring to that when he says magic numbers. If he does, I strongly agree :)

[the remark this answers is not in the document]
Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemRPMNightlyRelease.groovy]: ^^ this is exactly what I'm getting at.

[the remark this answers is not in the document]
Comment (ehelms) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemRPMNightlyRelease.groovy]: OK, I'll change it. My request for next time is to just state this from the beginning :) I find a suggested change or a question more helpful than cryptic (or magic ;)) text.

[the remark this answers is not in the document]
Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/rubygemRPMNightlyRelease.groovy]: Apologies, I thought it was fairly well-known what magic numbers in the context of programming are.

**Comment by ehelms** [comment]
Added a variable to define ruby version

**Comment by komidore64** [comment]
i'd prefer `ruby_version`, but this is fine too.

**Pull request merged**